app/Main.hs

```diff
@@ -1,13 +1,25 @@
 {-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RecordWildCards #-}
 module Main where
 
 import Web.Spock
 import Web.Spock.Config
+import Control.Monad.IO.Class (liftIO)
+import Data.Yaml.Config (loadYamlSettings, useEnv)
+import Data.Aeson (parseJSON, FromJSON, withObject, (.:))
+
+data AppSettings = AppSettings { appPort :: Int }
+
+instance FromJSON AppSettings where
+  parseJSON = withObject "AppSettings" $ \o -> do
```

Comment:
Also, what's your rules of defining what functions should be anonymous and which not?

Maybe here it would be more readable to extract it as a separate function?